### PR TITLE
Support set PF_LOGGING_LEVEL & refine sdk logger usage

### PR DESCRIPTION
**Pull request by brynn-code**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/_pf_client.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 from .operations._local_azure_connection_operations import LocalAzureConnectionOperations
 from .operations._tool_operations import ToolOperations
 
-logger = LoggerFactory.get_logger(name=LOGGER_NAME, verbosity=logging.WARNING)
+logger = LoggerFactory.get_logger(name=__name__, verbosity=logging.WARNING)
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_pf_client.py:22]: why changed this ?

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow/promptflow/_utils/logger_utils.py [hunk 2]
```diff
@@ -166,8 +168,21 @@ def clear(self):
         self._context_var.set(None)
 
 
+valid_logging_level = {"CRITICAL", "FATAL", "ERROR", "WARN", "WARNING", "INFO", "DEBUG", "NOTSET"}
+
+
+def get_pf_logging_level(default=logging.INFO):
+    logging_level = os.environ.get(PF_LOGGING_LEVEL, logging.INFO)
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_utils/logger_utils.py:175]: add a changelog on this env variable

**Review by thy09 — DISMISSED** [review]

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow/promptflow/_cli/_pf/_config.py [hunk 3]
```diff
@@ -1,14 +1,14 @@
 import argparse
 import json
-import logging
 
 from promptflow._cli._params import add_param_set_positional, logging_params
 from promptflow._cli._utils import activate_action, list_of_dict_to_dict
 from promptflow._sdk._configuration import Configuration, InvalidConfigValue
 from promptflow._sdk._constants import LOGGER_NAME
 from promptflow._sdk._utils import print_red_error
+from promptflow._utils.logger_utils import LoggerFactory
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_cli/_pf/_config.py:9]: add change log and FAQ for this change

File: src/promptflow/promptflow/_sdk/_pf_client.py (hunk 1 above)
Comment (brynn-code) [reply src/promptflow/promptflow/_sdk/_pf_client.py]: reverted

**Review by wangchao1230 — APPROVED** [review]

**Review by YingChen1996 — APPROVED** [review]

**Pull request merged**